### PR TITLE
[Address32][3/3] Use 32 bytes Address

### PR DESCRIPTION
**Pull request by areshand**
## Motivation

use 32 byte address 
disabled a list of tests and recorded in this issue https://github.com/aptos-labs/aptos-core/issues/256

**This diff is also stacked on top of PR https://github.com/aptos-labs/aptos-core/pull/310**

### Have you read the [Contributing Guidelines on pull requests](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan
cargo test
cargo x lint && cargo xfmt && cargo xclippy --all-targets

### CODE REVIEW

File: types/src/unit_tests/canonical_serialization_examples.rs [hunk 1]
```diff
@@ -1,264 +0,0 @@
...
-    transaction::{
-        ChangeSet, RawTransaction, Script, TransactionArgument, TransactionPayload, WriteSetPayload,
-    },
-    write_set::{WriteOp, WriteSet, WriteSetMut},
-};
-use bcs::to_bytes;
-
-#[test]
-fn test_access_path_canonical_serialization_example() {
-    let account_address = AccountAddress::new([
```
Comment (zekun000) [inline types/src/unit_tests/canonical_serialization_examples.rs:20]: we probably should fix it instead of deleting

File: types/src/event.rs [hunk 2]
```diff
@@ -17,7 +15,6 @@ use std::{
 /// A struct that represents a globally unique id for an Event stream that a user can listen to.
 /// By design, the lower part of EventKey is the same as account address.
 #[derive(Clone, Copy, Eq, Hash, Ord, PartialEq, PartialOrd)]
-#[cfg_attr(any(test, feature = "fuzzing"), derive(Arbitrary))]
```
Comment (zekun000) [inline types/src/event.rs:20]: why this can't be derived now?

File: sdk/transaction-builder/src/error_explain.rs [hunk 3]
```diff
@@ -36,6 +36,7 @@ mod tests {
     use move_core_types::{account_address::AccountAddress, ident_str};
 
     #[test]
+    #[ignore] // re-enable after cleaning up hardcoded code
```
Comment (zekun000) [inline sdk/transaction-builder/src/error_explain.rs:39]: this should work?

File: aptos-move/framework/core/sources/Account.move [hunk 4]
```diff
@@ -125,6 +125,7 @@ module CoreFramework::Account {
         authentication_key_prefix: vector<u8>,
         _witness: &T,
     ): (signer, vector<u8>) {
+        authentication_key_prefix = Vector::empty<u8>();
```
Comment (zekun000) [inline aptos-move/framework/core/sources/Account.move:128]: this is too hacky, we should cleanup the interface

File: types/src/unit_tests/canonical_serialization_examples.rs (hunk 1 above)
Comment (areshand) [reply types/src/unit_tests/canonical_serialization_examples.rs:20]: fixed.

File: types/src/event.rs (hunk 2 above)
Comment (areshand) [reply types/src/event.rs:20]: it doesn't support 40 bytes array.

File: sdk/transaction-builder/src/error_explain.rs (hunk 3 above)
Comment (areshand) [reply sdk/transaction-builder/src/error_explain.rs:39]: It fails here https://github.com/aptos-labs/aptos-core/blob/main/sdk/transaction-builder/src/error_explain.rs#L22, when loading a compile error map.  This error map is not updated when releasing the framework

File: aptos-move/framework/core/sources/Account.move (hunk 4 above)
Comment (areshand) [reply aptos-move/framework/core/sources/Account.move:128]: Yes, this will be in the follow up issue https://github.com/aptos-labs/aptos-core/issues/246

**Comment by zekun000** [comment]
need to rebuild the experimental artifacts too, `cargo run -- --package experimental` in framework crate

File: api/goldens/aptos_api::tests::events_test::test_get_events.json [hunk 5]
```diff
@@ -1,10 +1 @@
...
-  {
-    "key": "0x04000000000000000000000000000000000000000a550c18",
-    "sequence_number": "0",
-    "type": "0x1::Reconfiguration::NewEpochEvent",
-    "data": {
-      "epoch": "1"
-    }
-  }
-]
+[]
```
Comment (zekun000) [inline api/goldens/aptos_api::tests::events_test::test_get_events.json:1]: this seems wrong

File: api/src/tests/accounts_test.rs [hunk 6]
```diff
@@ -157,6 +153,7 @@ async fn test_get_account_resources_by_invalid_ledger_version() {
 // figure out a working module code, no idea where the existing one comes from
 #[ignore]
 #[tokio::test]
+#[ignore] // TODO(issue 81): re-enable after cleaning up the compiled code in the test
```
Comment (zekun000) [inline api/src/tests/accounts_test.rs:156]: it's duplicated ignore

File: api/src/tests/transactions_test.rs [hunk 7]
```diff
@@ -125,6 +125,7 @@ async fn test_get_transactions_output_user_transaction_with_script_function_payl
 // TODO: figure out correct module payload
 #[ignore]
 #[tokio::test]
+#[ignore] // re-enable after clean up the compiled code
```
Comment (zekun000) [inline api/src/tests/transactions_test.rs:128]: duplicate ignore

File: api/src/tests/transactions_test.rs [hunk 8]
```diff
@@ -925,6 +927,7 @@ async fn test_get_txn_execute_failed_by_missing_script_function_arguments() {
 }
 
 #[tokio::test]
+#[ignore] // re-enable after the sw
```
Comment (zekun000) [inline api/src/tests/transactions_test.rs:930]: why this is ignored? what's sw?

File: api/src/tests/events_test.rs [hunk 9]
```diff
@@ -4,13 +4,14 @@
 use crate::{current_function_name, tests::new_test_context};
 use percent_encoding::{utf8_percent_encode, NON_ALPHANUMERIC};
 
+static EVENT_KEY: &str =
```
[the remark this answers is not in the document]
Comment (areshand) [reply api/src/tests/events_test.rs:7]: good catch

File: api/src/tests/transactions_test.rs (hunk 8 above)
Comment (areshand) [reply api/src/tests/transactions_test.rs:930]: removed this ignore.

File: api/goldens/aptos_api::tests::events_test::test_get_events.json (hunk 5 above)
Comment (areshand) [reply api/goldens/aptos_api::tests::events_test::test_get_events.json:1]: updated the address and regenerate the golden file

File: api/src/tests/accounts_test.rs (hunk 6 above)
Comment (areshand) [reply api/src/tests/accounts_test.rs:154]: removed

File: aptos-move/aptos-transactional-test-harness/tests/diem_test_harness/example.exp [hunk 10]
```diff
@@ -1,17 +1,26 @@
 processed 10 tasks
 
+task 2 'publish'. lines 19-30:
+Error: Transaction discarded. VMStatus: status INVALID_AUTH_KEY of type Validation
```
Comment (zekun000) [inline aptos-move/aptos-transactional-test-harness/tests/diem_test_harness/example.exp:4]: I think those are unexpected

**Review by zekun000 — COMMENTED** [review]
there's some unexpected test results with INVALID_AUTH_KEY in *.exp files

File: aptos-move/aptos-transactional-test-harness/tests/diem_test_harness/example.exp (hunk 10 above)
Comment (areshand) [reply aptos-move/aptos-transactional-test-harness/tests/diem_test_harness/example.exp]: After debugging, I concatenate the auth_prefix + account_address to generate the new account address to make these tests work in 32-byte address. will followup to update the move code to completely remove the authkey_prefix in a separate PR

File: storage/aptosdb/src/state_store/state_store_test.rs [hunk 11]
```diff
@@ -189,16 +189,16 @@ fn test_retired_records() {
             (address3, value3.clone()),
         ],
         1, /* version */
-        4, /* expected_nodes_created */
+        3, /* expected_nodes_created */
```
Comment (areshand) [inline storage/aptosdb/src/state_store/state_store_test.rs:192]: @sitalkedia could you pls take a look to see if the change makes sense?

Comment (msmouse) [reply storage/aptosdb/src/state_store/state_store_test.rs:192]: Yeah I think it's good.

Comment (sitalkedia) [reply storage/aptosdb/src/state_store/state_store_test.rs:192]: +1. Changes look sane to me.

**Review by zekun000 — APPROVED** [review]
awesome, congrats!

**Pull request closed without merging**